pilot/pkg/config/kube/ingress/controller.go

```diff
@@ -185,14 +193,19 @@ func (c *controller) List(typ, namespace string) ([]model.Config, error) {
 
 		switch typ {
 		case model.VirtualService.Type:
-			_, virtualServices := ConvertIngressV1alpha3(*ingress, c.domainSuffix)
-			out = append(out, virtualServices)
+			ConvertIngressVirtualService(*ingress, c.domainSuffix, ingressByHost)
 		case model.Gateway.Type:
-			gateways, _ := ConvertIngressV1alpha3(*ingress, c.domainSuffix)
+			gateways := ConvertIngressV1alpha3(*ingress, c.domainSuffix)
```

Comment:
same comment as above. and you can get rid fo the convertIngressV1alpha3 as well